python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -50,26 +51,53 @@ def __init__(self, function, args=(), kwargs={},
         self.thread = None
         self.description = ''
 
+def getNumAliveThreads():
+    num_currently_running_command = 0
+    for this_thread in ThreadPool:
+        if this_thread._currently_running_command:
+            num_currently_running_command += 1
+    return num_currently_running_command
+
+def checkHeartBeat():
+
+    latest_timeNow = time.time()
+
+    for this_thread in ThreadPool:
+        thread_name = this_thread._thread_name
+        if heartbeat_times[thread_name] > 0:
+            last_time = heartbeat_times[thread_name]
+        else:
+            last_time = global_start_time
```

Comment:
Instead of setting `heartbeat_times[thread_name] = -1` and then checking this and setting it to `global_start_time` if it's negative, make `heartbeat_times = defaultdict(lambda: global_start_time)` and just ask it for last seen times. If it's never been asked then it will return the `global_start_time`, otherwise it will return the true last seen time.

I'm not completely sure but I think it should work and would remove some lines of code.